lib/pos_snapshot_entries.go

```diff
@@ -324,13 +334,21 @@ func (bav *UtxoView) GetSnapshotValidatorSetByStakeAmount(limit uint64) ([]*Vali
 // ordered by stake. This is useful when we need to know all the validators that
 // are in a leader schedule.
 func (bav *UtxoView) GetAllSnapshotValidatorSetEntriesByStake() ([]*ValidatorEntry, error) {
-	snapshotGlobalParams, err := bav.GetSnapshotGlobalParamsEntry()
+	snapshotGlobalParams, err := bav.GetCurrentSnapshotGlobalParamsEntry()
 	if err != nil {
 		return nil, errors.Wrapf(err, "GetAllSnapshotValidatorSetEntriesByStake: problem getting SnapshotGlobalParamsEntry: ")
 	}
 	return bav.GetSnapshotValidatorSetByStakeAmount(snapshotGlobalParams.ValidatorSetMaxNumValidators)
 }
 
+func (bav *UtxoView) GetAllSnapshotValidatorSetEntriesByStakeAtSnapshotAtEpochNumber(snapshotAtEpochNumber uint64) ([]*ValidatorEntry, error) {
```

Comment:
nit: no need for the second "Snapshot"
```suggestion
func (bav *UtxoView) GetAllSnapshotValidatorSetEntriesByStakeAtEpochNumber(snapshotAtEpochNumber uint64) ([]*ValidatorEntry, error) {
```